lib/services/elasticsearch.js

```diff
@@ -153,519 +166,479 @@ class ElasticSearch extends Service {
+      .then(({ body }) => {
+        const apiResponse = {
+          count: body.count
+        };
+
+        return apiResponse;
+      })
+      .catch(error => this.esWrapper.reject(error));
   }
 
   /**
    * Sends the new document to elasticsearch
    * Cleans data to match elasticsearch specifications
    *
-   * @param {Request} request
-   * @returns {Promise} resolve an object that contains _id
+   * @param {string} index - Index name
+   * @param {string} collection - Collection name
+   * @param {object} content - Document content
+   * @param {object} options - id (null), refresh (null), userId (null)
+   *
+   * @returns {Promise} { _id, _version, _source }
    */
-  create(request) {
-    const esRequest = initESRequest(request, ['refresh']);
-
-    esRequest.id = request.input.resource._id;
-    esRequest.body = request.input.body;
+  create (index, collection, content, { id=null, refresh=null, userId=null }) {
+    assertIsObject(content);
+
+    const esRequest = {
+      index: this._getESIndex(index, collection),
+      body: content,
+      id,
+      refresh
+    };
 
     assertNoRouting(esRequest);
     assertWellFormedRefresh(esRequest);
 
-    return this.kuzzle.indexCache.exists(esRequest.index, esRequest.type)
+    return this.kuzzle.indexCache.exists(index, collection)
       .then(exists => {
         if (! exists) {
           this.throw(
             'index_or_collection_does_not_exist',
-            esRequest.index,
-            esRequest.type
-          );
+            index,
+            collection);
         }
 
         // Add metadata
         esRequest.body._kuzzle_info = {
-          author: this._getUserId(request),
+          author: String(userId),
           createdAt: Date.now(),
           updatedAt: null,
-          updater: null,
-          active: true,
-          deletedAt: null
+          updater: null
         };
 
-        if (esRequest.id) {
-          // Check if the document exists and has not been deleted (active: false)
-          return this.client.get(
-            {index: esRequest.index,
-              type: esRequest.type,
-              id: esRequest.id}
-          )
-            .then(result => {
-              if (result._source._kuzzle_info &&
-                !result._source._kuzzle_info.active) {
-                // The document is inactive, we replace it and masquerade the result as a creation
-                return this.client.index(esRequest)
-                  .then(res => {
-                    res.result = 'created';
-                    res.created = true;
-
-                    return this.refreshIndexIfNeeded(
-                      esRequest,
-                      _.extend(res, {_source: request.input.body}));
-                  })
-                  .catch(error => Bluebird.reject(
-                    this.esWrapper.formatESError(error))
-                  );
-              }
+        const promise = esRequest.id
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "promise". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1488)